test/functional/pdo_sqlsrv/pdo_ae_output_param_decimals.phpt

```diff
@@ -0,0 +1,241 @@
+            // 1 digit ('.') + possibly the negative sign
+            $len = $precision + 2 + $i;
+        } elseif ($scale > 0) {
+            // compare up to $precision + 1 digit ('.') 
+            // + possibly the negative sign
+            $len = $precision + 1 + $i;
+        } else {
+            // in this case, $scale = 0 
+            // compare up to $precision + possibly the negative sign
+            $len = $precision + $i;
+        }
+        
+        trace("Comparing $len...");
+        $result = substr_compare($inputStr, $fetchedStr, 0, $len);
+        if ($result != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+function testOutputDecimals($inout) 
+{
+    global $pdoParamTypes, $dataTypes, $inputValuesInit, $precisions, $inputPrecision, $errors;
+    
+    try {
+        $conn = connect();
+        
+        $tbname = "test_decimals_types";
+        $spname = "test_decimals_proc";
+
+        foreach ($dataTypes as $dataType) {
+            foreach ($precisions as $precision => $scales) {
+                foreach ($scales as $scale) {
+                    // construct the input values depending on the precision and scale
+                    $precDiff = $inputPrecision - ($precision - $scale);
+                    $inputValues = $inputValuesInit;
+                    foreach ($inputValues as &$inputValue) {
+                        $inputValue = $inputValue / pow(10, $precDiff);
+                    }
+                    
+                    $type = "$dataType($precision, $scale)";
+                    trace("\nTesting $type:\n");
+                    
+                    //create and populate table
+                    $colMetaArr = array(new ColumnMeta($type, "c_det"), new ColumnMeta($type, "c_rand", null, "randomized"));
+                    createTable($conn, $tbname, $colMetaArr);
+
+                    $stmt = insertRow($conn, $tbname, array("c_det" => $inputValues[0], "c_rand" => $inputValues[1] ), null, $r);
+                    
+                    // fetch with PDO::bindParam using a stored procedure
+                    dropProc($conn, $spname);
+                    $spSql = "CREATE PROCEDURE $spname (
+                                    @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                    SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                    $conn->query($spSql);
+            
+                    // call stored procedure
+                    $outSql = getCallProcSqlPlaceholders($spname, 2);
+                    foreach ($pdoParamTypes as $pdoParamType) {
+                        $det = $rand = 0.0;
+                        $stmt = $conn->prepare($outSql);
+                    
+                        $len = 2048;
+                        if ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                            $len = PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE;
+                            $det = $rand = 0;

```

Comment:
perhaps for PDO::PARAM_STR and PDO::PARAM_LOB you want to specify $det = $rand = ""; as well